File: backend/app/models/reflections.py
"""Reflection schemas (Phase 2.5, REQ-8/9/12)."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field

ReflectionKind = Literal["momentum", "drift", "theme"]
ReflectionStatus = Literal["active", "pinned", "dismissed"]

BODY_MAX = 2_000


class ReflectionOut(BaseModel):
    id: str
    kind: ReflectionKind
    body: str
    evidence: dict
    status: ReflectionStatus
    created_at: str


class ReflectionPatch(BaseModel):
    status: ReflectionStatus


class GenerateOut(BaseModel):
    reflections: list[ReflectionOut]
    reason: str | None = None

File: frontend/src/lib/reflections.ts
import { api } from './api'

export type ReflectionKind = 'momentum' | 'drift' | 'theme'
export type ReflectionStatus = 'active' | 'pinned' | 'dismissed'

export interface ReflectionEvidence {
  objective_ids?: string[]
  journal_entry_ids?: string[]
  terms?: string[]
}

export interface Reflection {
  id: string
  kind: ReflectionKind
  body: string
  evidence: ReflectionEvidence
  status: ReflectionStatus
  created_at: string
}

export interface GenerateResult {
  reflections: Reflection[]
  reason: string | null
}

export const KIND_COLOR: Record<ReflectionKind, string> = {
  momentum: 'var(--aqua)',
  drift: 'var(--warn)',
  theme: 'var(--node-ctx)',
}

export async function fetchReflections(status?: ReflectionStatus): Promise<Reflection[]> {
  const { data } = await api.get<Reflection[]>('/reflections', {
    params: status ? { status } : undefined,
  })
  return data
}

export async function generateReflections(): Promise<GenerateResult> {
  const { data } = await api.post<GenerateResult>('/reflections/generate')
  return data
}

export async function patchReflection(
  id: string,
  status: ReflectionStatus,
): Promise<Reflection> {
  const { data } = await api.patch<Reflection>(`/reflections/${id}`, { status })
  return data
}
